refactor: Community 전체 수정

daehwan2yo opened the pull request:
- User 에 UserNotFound Exception 추가
- Test 에 PostFactory 추가
- 댓글수, 조회수 추가
- 빌드 성공

Resolves: #16

daehwan2yo committed:
refactor: Community 전체 수정
- User 에 UserNotFound Exception 추가
- Test 에 PostFactory 추가
- 댓글수, 조회수 추가
- 빌드 성공

Resolves: #16

always0ne commented on src/main/java/com/jukbang/api/community/contorller/CommentsController.java:
```diff
@@ -6,85 +6,88 @@
 @CrossOrigin(origins = "*")
 @RequiredArgsConstructor
-@RequestMapping("/community/comments/{univId}")
+@RequestMapping("/community/{univId}")
```
여기에 /{postId}/comments 추가해도 좋아
그런데 댓글이랑 게시글 따로받는거 유지하기로한거야?

gyedaehwan replied:
엉엉 댓글 게시글 따로 받는걸루 했엉

always0ne commented on src/main/java/com/jukbang/api/community/contorller/PostController.java:
```diff
@@ -15,39 +18,39 @@
     @GetMapping
     public List getPostList() {
-        return communityService.getCommunitylist();
+        return postService.getPostList();
```
이거 Pagenation 적용해보자
같이할 때 설명해줄게

always0ne commented on src/main/java/com/jukbang/api/community/entity/Post.java:
```diff
@@ -16,24 +16,48 @@
+        this.body= body;
+    }
+
+    public void setViews(int views){
+        this.views = views;
+    }
+
+    public void setComments(int comments) { this.comments = comments; }
```
보통 값이 하나씩 증가하거나 감소시키는 경우니깐
increase views
increase comments
decrease comments
이렇게 구분하는게 어떨까

gyedaehwan replied:
오키 맞춰서 수정할게

always0ne commented on src/main/java/com/jukbang/api/community/service/CommentsService.java:
```diff
@@ -1,128 +1,146 @@
-                    .body(boardEntity.getBody())
-                    .univId(boardEntity.getUnivId())
+    public List<CommentsDto> getCommentsList(Long postId) {
+        List<Comments> commentsEntities = commentsRepository.findAllByPostId(postId);
+
+        List<CommentsDto> commentsDtoList = new ArrayList<>();
+
+        for (Comments commentsEntity : commentsEntities) {
+            CommentsDto commentsDTO = CommentsDto.builder()
```
이부분도 projection 사용하면 훨씬 더 간단하고 쿼리 최적화게 구현할수 있어

always0ne commented on src/main/java/com/jukbang/api/community/service/CommentsService.java:
```diff
@@ -1,128 +1,146 @@
+                            CreateCommentRequest createCommentRequest ) {
+
+        // 존재하는 Id 인지 확인
+        userRepository.findByUserId(userId).orElseThrow(UserNotFoundException::new);
```
레포지터리에 있는지 없는지여부만 구할수 있는 기능있는데 그거사용하면 좋을거같아

gyedaehwan replied:
회의하면서 방법좀 공유해주세용.. ㅎㅎ

always0ne commented on src/main/java/com/jukbang/api/community/service/PostService.java:
```diff
@@ -0,0 +1,186 @@
+        return postListDto_List;
+    }
+/*
+    커뮤니티 내 내 게시글 불러오기 기능 => 프런트 ppt 에 없음
+    1. 게시글 리스트 형식으로 할지?
+    2. 게시글 상세정보 형식으로 할지?
```
다같이 회의할때 이야기해보자

always0ne reviewed (commented):
우선 내가 올린 방 리팩토링 코드랑 살짝 겹치는부분이 있어서 내꺼 합치고 한번 손봐야할거같아
그리고 테스트코드도 다시 활성화 시켜서 빌드테스트하는거까지 하면 좋을거같아

always0ne committed:
Merge branch 'develop' into 16-community

always0ne committed:
Merge remote-tracking branch 'origin/develop' into 16-community

always0ne committed:
refactor: after merge from  develop  to 16-community

daehwan2yo committed:
build 성공,
test 에서 자꾸 runtime error 인데 commentRepository 에서 객체 불러오는게 오류인거 같은데 어떻게 해야할지 모르겠음.

always0ne committed:
Merge branch 'develop' into 16-community

always0ne commented on src/main/java/com/jukbang/api/community/repository/CommentsRepository.java:
```diff
@@ -5,8 +5,11 @@
 
 @Repository
 public interface CommentsRepository extends JpaRepository<Comments, Long> {
+    List<Comments> findAllByPostId(Long postId);
 
+    Optional<Comments> findByPostIdAndCommentsId(Long postId, Long commentsId);
 }
```
이 부분 JPA규칙에 맞지 않는 인터페이스여서 에러 발생

always0ne commented on src/test/java/com/jukbang/api/community/CreateComment.java:
```diff
@@ -16,33 +19,24 @@
 public class CreateComment extends BaseControllerTest {
 
     @Autowired
-    private CommunityService communityService;
+    private CommentsService commentsService;
 
```
안쓰는 서비스 제거 필요

always0ne commented on src/test/java/com/jukbang/api/community/CreateComment.java:
```diff
@@ -16,33 +19,24 @@
 
-        Long postId = communityService.SavePost(1,createPostRequest);
         CreateCommentRequest createCommentRequest = CreateCommentRequest.builder()
-                .id(1)
-                .writer("writer")
-                .body("body")
+                .body("TestComments")
                 .build();
 
-
-        this.mockMvc.perform(RestDocumentationRequestBuilders.post("/community/comments/{univId}/{postId}", 1, postId)
+        this.mockMvc.perform(RestDocumentationRequestBuilders.post("/community/{univId}/{postId}/comments", 1, postId)
```
권한 문제 발생하니 signUpUser사용해서 Authorization  헤더 추가할것
자세한건 Room쪽 테스트 코드 보면 될거야

always0ne reviewed (commented):
이정도 고치면 빌드하는데 이상없을거야

daehwan2yo committed:
build success,
403 error,
2개의 test 는 200 (GetPost, GetPostList)

daehwan2yo committed:
test : user 관련 부분 signUpUser사용해서 Authorization 헤더 추가

java : Post 객체에서 List<> 삭제 (OneToMany 삭제)  -> Comment 객체에서 ManyToOne만 유지 , commentsRepository에서 findByAll 로 리스트 불러옴

daehwan2yo committed:
community test complete

daehwan2yo committed:
community test delete update

daehwan2yo committed:
community devide role (student/all)
error : test/ getpost, getpostlist

daehwan2yo committed:
community divide complete

always0ne committed:
Merge branch 'develop' into 16-community

always0ne reviewed (approved):
굿
